### ISSUE TITLE
Block Params (Stage 1)

### ISSUE THREAD

**Issue by hzoo** [babel/proposals#53]
https://github.com/samuelgoto/proposal-block-params

**Comment by littledan** [comment]
Note that this proposal led to significant concerns from implementers when it was presented at TC39. I imagine it could be nontrivial to implement in Babel for the same reasons.

**Comment by nicolo-ribaudo** [comment]
I don't think that it would be difficult; we would reuse the same logic we already have for disambiguating async arrows.

**Comment by littledan** [comment]
I don't mean in parsing, but to make it so that return, break and continue inside block parameters do so for the outer function.

**Comment by hzoo** [comment]
Yeah, I just wanted to make an issue for each proposal but most of them we probably don't want to implement yet.

We need to automate this work lol, it's always out of date of course
